src/useFetch.ts

```diff
@@ -83,13 +83,17 @@ function useFetch<TData = any>(...args: UseFetchArgs): UseFetch<TData> {
       const isCached = cache.has(requestID)
       const cachedData = cache.get(requestID)
       if (isCached && cachePolicy === CACHE_FIRST) {
+        setLoading(true)
         const whenCached = cache.get(requestID + ':ts')
         let age = Date.now() - whenCached
         if (cacheLife > 0 && age > cacheLife) {
           cache.delete(requestID)
           cache.delete(requestID + ':ts')
         } else {
-          return cachedData
+          res.current.data = cachedData
+          data.current = cachedData
+          setLoading(false)
+          return data.current
```

Comment:
beautiful, we just need to not modify the state here. Unless for some reason we expect `loading` to be true for some reason (other than setting it to true above in your code change)

Reply:
Once these things are resolved, I'll most likely merge this immediately.

Reply:
Amazing work so far btw

Reply:
The only reason I chose to toggle the `loading` state here is because if we don't modify the state, the hook never re-renders and so we cant listen for updates to data out the other end.

Reply:
Hopefully that makes sense

Reply:
ah! I see, hm.. let me think about this